mmv1/third_party/terraform/services/logging/data_source_google_logging_folder_settings.go

```diff
@@ -0,0 +1,100 @@
+package logging
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+func DataSourceGoogleLoggingFolderSettings() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGoogleLoggingFolderSettingsRead,
+		Schema: map[string]*schema.Schema{
+			"folder": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The folder for which to retrieve settings.`,
+			},
+			"disable_default_sink": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: `If set to true, the _Default sink in newly created projects and folders will created in a disabled state. This can be used to automatically disable log storage if there is already an aggregated sink configured in the hierarchy. The _Default sink can be re-enabled manually if needed.`,
+			},
+			"kms_key_name": {
+				Type:     schema.TypeString,
+				Optional: true,
```

Comment:
I think this is Computed, given this is a datasource? It's not required for a getSettings call: https://cloud.google.com/logging/docs/reference/v2/rest/v2/folders/getSettings

(same for both other levels)